src/main/java/org/folio/readingroom/service/impl/ReadingRoomServiceImpl.java

```diff
@@ -54,14 +54,26 @@ public ReadingRoom updateReadingRoom(UUID readingRoomId, ReadingRoom readingRoom
       throw new IdMismatchException(
         "The ID provided in the request URL does not match the ID of the resource in the request body");
     }
-    var existingEntity = readingRoomRepository.findById(readingRoomId)
-      .orElseThrow(() ->
-        new NotFoundException(String.format("Reading room with id %s doesn't exists", readingRoomId)));
+    var existingEntity = getReadingRoomById(readingRoomId);
     validateServicePoints(readingRoomDto.getServicePoints(), readingRoomId);
     updateModifiedFields(existingEntity, readingRoomMapper.toEntity(readingRoomDto));
     return readingRoomMapper.toDto(readingRoomRepository.save(existingEntity));
   }
 
+  @Override
+  public void deleteReadingRoomById(UUID readingRoomId) {
+    log.debug("deleteReadingRoomById:: soft deleting reading room with id {}", readingRoomId);
+    ReadingRoomEntity readingRoomEntity = getReadingRoomById(readingRoomId);
+    readingRoomEntity.setDeleted(true);
+    readingRoomRepository.save(readingRoomEntity);
+  }
```

Comment:
What will happen to the patrons that have allowed access to this reading-room that is being marked to is deleted ? Will that not be updated ?